Randolph-dev/Druid_Partnering_Project_React24k
# Scrum 144 mautic personalised content with frontend script

## Related ticket:
[SCRUM-144](https://druidpartneringproject2024bch.atlassian.net/browse/SCRUM-144?atlOrigin=eyJpIjoiY2U2MmEzZDc4MTRkNDYzYTg5NGM3MmM3ZjYwYjUzYmYiLCJwIjoiaiJ9)

## In this PR:
- Create a custom form to send user type to Mautic
- Fetch the user type from mautic
- Render personalised contents on the Frontpage
- Add 3 more frontpages for different user types

## Testing instruction
- Checkout branch
- Backend: `lando composer I`, `lando db-import 1.9.gz`, `lando drush cim`
- Frontend: add 2 more env variable in this format:
```shell
VITE_DRUPAL_REST=https://druidpartneringapp.lndo.site/api/
VITE_MAUTIC_API_TOKEN=API_TOKEN_HERE
```
- Go to front page, see the current content
- Go to contact page, fill in the form, choose something other than Feedback
- Go back to frontage, refresh and you should see the content of the front page changed

[SCRUM-144]: https://druidpartneringproject2024bch.atlassian.net/browse/SCRUM-144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
